platform/security/idp/security-idp-plugin/security-idp-plugin-nameidpolicy/src/test/groovy/org/codice/ddf/security/idp/plugin/nameidpolicy/NameIdPolicyPresignPluginSpec.groovy

```diff
@@ -0,0 +1,417 @@
+        NameID.PERSISTENT | NameID.EMAIL        | false
+    }
+
+    @Unroll("transient NameIDPolicy Format tests for Format #format")
+    def 'transient NameIDPolicy Format tests'() {
+        setup:
+        nameIdMock.setFormat(format)
+        nameIdPolicyMock.getFormat() >> policyFormat
+
+        when:
+        plugin.processPresign(responseMock, authnRequestMock, spMetadataMock, supportedBindingsMock)
+
+        then:
+        if (!isError) {
+            assert (nameIdMock.value == EXAMPLE_NAME_ID_VALUE && nameIdMock.format == policyFormat)
+            return
+        }
+        thrown(UnsupportedOperationException)
+
+        where:
+        policyFormat     | format              | isError
+        NameID.TRANSIENT | null                | false
+        NameID.TRANSIENT | NameID.UNSPECIFIED  | false
+        NameID.TRANSIENT | NameID.PERSISTENT   | false
+        NameID.TRANSIENT | NameID.TRANSIENT    | false
+        NameID.TRANSIENT | NameID.X509_SUBJECT | false
+        NameID.TRANSIENT | NameID.EMAIL        | false
+    }
+
+    @Unroll("x509 NameIDPolicy Format tests for Format #format")
+    def 'x509 NameIDPolicy Format tests'() {
+        setup:
+        nameIdMock.setFormat(format)
+        nameIdPolicyMock.getFormat() >> policyFormat
+
+        when:
+        plugin.processPresign(responseMock, authnRequestMock, spMetadataMock, supportedBindingsMock)
+
+        then:
+        if (!isError) {
+            assert (nameIdMock.value == EXAMPLE_NAME_ID_VALUE && nameIdMock.format == policyFormat)
+            return
+        }
+        thrown(UnsupportedOperationException)
+
+        where:
+        policyFormat        | format              | isError
+        NameID.X509_SUBJECT | null                | true
+        NameID.X509_SUBJECT | NameID.UNSPECIFIED  | true
+        NameID.X509_SUBJECT | NameID.PERSISTENT   | true
+        NameID.X509_SUBJECT | NameID.TRANSIENT    | true
+        NameID.X509_SUBJECT | NameID.X509_SUBJECT | false
+        NameID.X509_SUBJECT | NameID.EMAIL        | true
+    }
+
+    @Unroll("email NameIDPolicy Format tests for Format #format")
+    def 'email NameIDPolicy Format tests'() {
+        setup:
+        nameIdMock.setFormat(format)
+        nameIdPolicyMock.getFormat() >> policyFormat
+
+        when:
+        plugin.processPresign(responseMock, authnRequestMock, spMetadataMock, supportedBindingsMock)
+
+        then:
+        if (!isError) {
+            assert (nameIdMock.value == EXAMPLE_EMAIL && nameIdMock.format == policyFormat)
+            return
+        }
+        thrown(UnsupportedOperationException)
+
+        where:
+        policyFormat | format              | isError
+        NameID.EMAIL | null                | false
+        NameID.EMAIL | NameID.UNSPECIFIED  | false
+        NameID.EMAIL | NameID.PERSISTENT   | false
+        NameID.EMAIL | NameID.TRANSIENT    | false
+        NameID.EMAIL | NameID.X509_SUBJECT | false
+        NameID.EMAIL | NameID.EMAIL        | false
+    }
```

Comment:
❓ Could you just combine all of these identical tests into one with one big table? (excluding the x509 test)

Reply:
I could group the tests a little more. But if this plugin becomes a little more fleshed out, then the current layout is much better. It allows for custom logic to be put in for each type of `NameIDPolicy` Format. Like the `X509SubjectName` stuff.